daffodil-core/src/main/scala/org/apache/daffodil/compiler/Compiler.scala

```diff
@@ -99,6 +99,8 @@ final class ProcessorFactory private(
       new SchemaSet(optRootSpec, schemaSource, validateDFDLSchemas, checkAllTopLevel, tunables,
         compilerExternalVarSettings))
 
+  def walkDSOM[T](walker: AbstractDSOMWalker[T]): T = walker.walkFromRoot(sset.root)
```

Comment:
Normally one would call ``processorFactory.onPath(..)``, which asserts that there are no errors and I think can result in finishing some lazy compilation? Two questions, more directed at Mike who I think knows more about the internal compialtion steps than me:
1. Should we also assert here that there are no errors?
2. Do we need to also do something to account for lazy compilation, such as ExecutionMode.usingCompilerMode? Is it possible for an SDE to get hit while walking and performing lazy compliation, or should all that be finished by the time onPath is called?